flutter_local_notifications_windows/lib/src/plugin/ffi.dart

```diff
@@ -0,0 +1,196 @@
+import "dart:ffi";
+import "package:ffi/ffi.dart";
+
+import "../details.dart";
+import "../ffi/bindings.dart";
+import "../ffi/utils.dart";
+
+import "base.dart";
+
+void _globalLaunchCallback(NativeLaunchDetails details) {
+  FlutterLocalNotificationsWindows.instance?._onNotificationReceived(details);
+}
+
+extension on String {
+  bool get isValidGuid => length == 36
+    && this[8] == "-"
+    && this[13] == "-"
+    && this[18] == "-"
+    && this[23] == "-";
+}
+
+/// The FFI implementation of `package:flutter_local_notifications`.
```

Comment:
Didn't know it's possible to reference a package so thanks for that. However, can this be updated to `The Windows implementation of [FlutterLocalNotificationsPlatform]` and then have a reference to the main plugin package in a separate paragraph (for example, like https://github.com/flutter/packages/blob/991c70177e8ff145f7d1c500d9dff86c6c7f92f4/packages/shared_preferences/shared_preferences_windows/lib/shared_preferences_windows.dart#L18)? Reason for pitching this is FFI  sounds too generic. I believe the Linux implementation could technically be said to use FFI as it depends on `dbus` package that in turn makes use of FFI. 

I just realised other parts of the plugin like the Linux one could do with similar changes so I'll note this to later as it's not tied to this PR

Reply:
> Didn't know it's possible to reference a package so thanks for that. 

Oh I don't think that does anything special in DartDocs, I just like `code style` in mardown. 

> However, can this be updated to The Windows implementation of [FlutterLocalNotificationsPlatform]

That would be `plugin/base.dart`

> Reason for pitching this is FFI sounds too generic. I believe the Linux implementation could technically be said to use FFI as it depends on dbus package that in turn makes use of FFI.

Right, but FFI crashes web builds. So I have `plugin/stub.dart` and `plugin/ffi.dart`, where the latter is the true, FFI-based implementation and the former is just a stub for web builds. `plugin/base.dart` is the base class that actually decides what the Windows plugin offers.

Reply:
> That would be plugin/base.dart

I get what you mean but what's there is an abstract class and how it's used is more like an interface. It's not exposed publicly nor can it be constructed for it to be an implementation

Reply:
Changed